data/src/main/java/org/apache/iceberg/data/BaseFileWriterFactory.java

```diff
@@ -150,8 +150,9 @@ protected BaseFileWriterFactory(
   public DataWriter<T> newDataWriter(
       EncryptedOutputFile file, PartitionSpec spec, StructLike partition) {
     EncryptionKeyMetadata keyMetadata = file.keyMetadata();
-    Map<String, String> properties = table.properties();
-    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+    Map<String, String> properties = table == null ? ImmutableMap.of() : table.properties();
+    MetricsConfig metricsConfig =
+        table == null ? MetricsConfig.getDefault() : MetricsConfig.forTable(table);
```

Comment:
need to handle null table, as a few use-cases for FileAppanderFactory uses it without an actual table